homeassistant/components/venstar/climate.py

```diff
@@ -165,12 +171,23 @@ def hvac_mode(self):
             return HVAC_MODE_AUTO
         return HVAC_MODE_OFF
 
+    @property
+    def hvac_action(self):
+        """Return current operation mode ie. heat, cool, auto."""
+        if self._client.state == self._client.STATE_IDLE:
+            return CURRENT_HVAC_IDLE
+        if self._client.state == self._client.STATE_HEATING:
+            return CURRENT_HVAC_HEAT
+        if self._client.state == self._client.STATE_COOLING:
+            return CURRENT_HVAC_COOL
+        return CURRENT_HVAC_OFF
+
     @property
     def fan_mode(self):
-        """Return the fan setting."""
-        if self._client.fan == self._client.FAN_AUTO:
-            return HVAC_MODE_AUTO
-        return STATE_ON
+        """Return the current fan mode."""
+        if self._client.fanstate == self._client.FANSTATE_ON:
+            return FAN_ON
```

Comment:
`fan_mode` should return the fan mode, not the fan state. Ie we should return one of the modes in `fan_modes` list.